src/genericobject.cpp

```diff
@@ -56,6 +58,10 @@ std::string gob_cmd_update_position(
 	writeV3F1000(os, acceleration);
 	// yaw
 	writeF1000(os, yaw);
+	// pitch
+	writeF1000(os, pitch);
+	// roll
+	writeF1000(os, roll);
```

Comment:
Only send pitch and roll here if automatic_rotate is above `0.001f` or `GenericCAO::processMessage` will be offset at reading the packet when the automatic rotation is disabled.

Reply:
Hmm. This kind of a hard one. Right now, on the client side in [content_cao.cpp](https://github.com/CoderForTheBetter/minetest/blob/31f38a23f7143a38738d8c0c35d8ef5b472acd52/src/content_cao.cpp#L1307) it is read like this due to automatic_rotate:
```
if (std::fabs(m_prop.automatic_rotate) < 0.001f) {
	m_yaw = readF1000(is);
	m_pitch = readF1000(is);
	m_roll = readF1000(is);
} else {
	readF1000(is);
}
```
I'll just point that out for now, so to accomplish what you're asking could I introduce a new parameter into gob_cmd_update_position() ([defined here, where we are talking about it](https://github.com/CoderForTheBetter/minetest/blob/31f38a23f7143a38738d8c0c35d8ef5b472acd52/src/genericobject.cpp#L39)) for automatic_rotate?

**To summerize:**
I want to add a parameter to gob_cmd_update_position() [here](https://github.com/CoderForTheBetter/minetest/blob/31f38a23f7143a38738d8c0c35d8ef5b472acd52/src/genericobject.cpp#L39) so that it takes on the form of:
```
std::string gob_cmd_update_position(
	v3f position,
	v3f velocity,
	v3f acceleration,
	f32 yaw,
	f32 pitch,
	f32 roll,
        f32 automatic_rotate
	bool do_interpolate,
	bool is_movement_end,
	f32 update_interval
)
(Note the new parameter "automatic_rotate." Hopfully f32 type is fine/good)
```
With the new parameter I can do this inside of gob_cmd_update_position():
```
...
// yaw
writeF1000(os, yaw);
if (automatic_rotate < 0.001f) {
	// pitch
	writeF1000(os, pitch);
	// roll
	writeF1000(os, roll);
}
...
```

The only places now that gob_cmd_update_position() is called from (has always been like this) is in the file content_sao.cpp in the function [LuaEntitySAO::sendPosition](https://github.com/CoderForTheBetter/minetest/blob/31f38a23f7143a38738d8c0c35d8ef5b472acd52/src/content_sao.cpp#L786) and in the function [PlayerSAO::step](https://github.com/CoderForTheBetter/minetest/blob/31f38a23f7143a38738d8c0c35d8ef5b472acd52/src/content_sao.cpp#L1117).

**The LuaEntitySAO::sendPosition call of gob_cmd_update_position will look like so:**
```
std::string str = gob_cmd_update_position(
	m_base_position,
	m_velocity,
	m_acceleration,
	m_yaw,
	m_pitch,
	m_roll,
	m_prop.automatic_rotate
	do_interpolate,
	is_movement_end,
	update_interval
);
```

**The PlayerSAO::step call of gob_cmd_update_position will look like so:**
```
std::string str = gob_cmd_update_position(
	pos,
	v3f(0.0f, 0.0f, 0.0f),
	v3f(0.0f, 0.0f, 0.0f),
	m_yaw,
	m_pitch,
	m_roll,
	m_prop.automatic_rotate
	true,
	false,
	update_interval
);
```
Should I just make commit so that you can review it and understand it easier, @SmallJoker?

Reply:
@SmallJoker Great! That is much more simple. This is great discussion if automatic_rotate should ever be added for pitch and roll. I may open an issue after this that refers to this thread. I will fix this and review all of my code once again for style issues, then I'll push everything.